src/Operations.jl

```diff
@@ -835,13 +838,11 @@ function download_artifacts(ctx::Context;
         pkg_root === nothing || push!(pkg_roots, pkg_root)
     end
     push!(pkg_roots, dirname(env.project_file))
-    used_artifact_tomls = Set{String}()
-    download_jobs = Function[]
+    download_jobs = Dict{SHA1, Function}()
 
     print_lock = Base.ReentrantLock() # for non-fancyprint printing
 
-    # name -> (state, bar) where state is :ready, :running, :done, or :failed
-    download_states = Dict{String, DownloadState}()
```

Comment:
Artifact names are not globally unique, so I changed the key to the artifact tree hash.